htdocs/core/db/DoliDB.class.php

```diff
@@ -30,7 +30,7 @@
 abstract class DoliDB implements Database
 {
 	/** @var string Force subclass to implement VERSIONMIN */
-	const VERSIONMIN=self::VERSIONMIN;
```

Comment:
@mdeweerd @eldy this make phpstan crash

Reply:
I opened an [issue with phpstan](https://github.com/phpstan/phpstan/issues/10155) .

I guess that for know this line has to be commented so that phpstan can pass.  It kept phan happy and also helps enforce that the constant is defined in derived classes though.

Reply:
I also found why the cache is not in '/tmp' in my project.

I have the [tmpDir option](https://phpstan.org/config-reference#caching) set:

```yaml
parameters:
  tmpDir: .cache/phpstan
```

If you rename phpstan.neon to phpstan.neon.dist, then you can set local parameters in phpstan.neon.  So for the github action you could create phpstan.neon with the specific location for the phpstan cache directory.
That helps avoid caching the entire /tmp directory which may not only hold phpstan cache.

Reply:
@mdeweerd phpstan bug has been fixed